### PR TITLE
add default async

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

File: libs/langchain/langchain/chains/base.py [hunk 1]
```diff
@@ -227,7 +227,9 @@ async def _acall(
             A dict of named outputs. Should contain all outputs specified in
                 `Chain.output_keys`.
         """
-        raise NotImplementedError("Async call not supported for this chain type.")
+        return await asyncio.get_running_loop().run_in_executor(
+            None, self._call, inputs, run_manager
```
Comment (eyurtsev) [inline libs/langchain/langchain/chains/base.py:231]: Are we able to remove the conditional from ainvoke so it's always relying on acall?

```
    async def ainvoke(
        self,
        input: Dict[str, Any],
        config: Optional[RunnableConfig] = None,
        **kwargs: Any,
    ) -> Dict[str, Any]:
        if type(self)._acall == Chain._acall:
            # If the chain does not implement async, fall back to default implementation
            return await asyncio.get_running_loop().run_in_executor(
                None, partial(self.invoke, input, config, **kwargs)
            )

```

Same question probably applies to a number of the other method implementations

**Review by eyurtsev — APPROVED** [review]

**Review by nfcampos — APPROVED** [review]

File: libs/langchain/langchain/chains/base.py (hunk 1 above)
Comment (nfcampos) [reply libs/langchain/langchain/chains/base.py:231]: Yes would be good to do for all the classes that now have default async implementation

**Review by eyurtsev — APPROVED** [review]
Looks good to me -- any simple tests for missing implementations would be great

**Comment by mkhludnev** [comment]
Hi could you clarify my understanding: are these defaults are less performant than using a specialized async clients. I'm asking in scope of new QdrandVectorStore https://github.com/langchain-ai/langchain/pull/24164#issuecomment-2343930026

**Pull request merged**